pulpcore/cli/rpm/remote.py

```diff
@@ -43,7 +53,23 @@ def remote(ctx: click.Context, pulp_ctx: PulpContext, remote_type: str) -> None:
     click.option(
         "--policy", type=click.Choice(["immediate", "on_demand", "streamed"], case_sensitive=False)
     ),
-    click.option("--sles-auth-token"),
+    pulp_option(
+        "--sles-auth-token",
+        allowed_with_contexts=(PulpRpmRemoteContext,),
+    ),
+    pulp_option(
+        "--uln-server-base-url",
+        default="https://linux-update.oracle.com/",
+        help=_("ULN Server base URL, default is 'https://linux-update.oracle.com/'"),
+        allowed_with_contexts=(PulpUlnRemoteContext,),
+    ),
+    pulp_option(
+        "--url",
```

Comment:
I have added the check for uln `url` as well as help_text. The only problem here now is we have two `url` options/fields in `--help` of uln remote. Is there a way to remove the base class url option? 

```
Options:
  --name TEXT                     [required]
  --url TEXT                      [required]
  --ca-cert TEXT                  a PEM encoded CA certificate or @file
                                  containing same
  .
  .
  .
  --policy [immediate|on_demand|streamed]
  --uln-server-base-url TEXT      ULN Server base URL, default is
                                  'https://linux-update.oracle.com/'
  --url TEXT                      Use the ULN channel name starting with
                                  uln:// here.  [required]
  --help                          Show this message and exit.
```

Reply:
Well, the other `--url` option comes via the `common_remote_{create,update}_options` which are plain lists.
So maybe we can create a filtered copy of these `[option for option in ... if option.name != "url"]`, and then add one `--url` option for the `RpmRemoteContext` and one for the `UlnRemoteContext`.
@gerrod3 may know whether they turn up correctly in the corresponding help outputs.

Reply:
The `common_remote_{create,update}_options` are lists of decorators that come from `click` so they don't have argument `name` or at least callable arguments.

Reply:
Then maybe it it impossible to use that shortcut here.

Reply:
So where do you suggest to add this condition? Adding it to the base class doesn't look like a good idea.

Reply:
Yeah, it's not so much a condition. I think we need to specify the used parameters explicitly here. Not particularly happy about this, but probably the only way to get this forward now.

Reply:
You mean to remove generic `common_remote_{create/update}_options` options and define them explicitly in `remote.py`?

Reply:
Yes, this is the way forward i can see today. And i know it's bad for consistency.